l1/ecl_l1_pos_controller.cpp

```diff
@@ -45,11 +45,27 @@
 using matrix::Vector2f;
 using matrix::wrap_pi;
 
-float ECL_L1_Pos_Controller::nav_roll()
+
+void ECL_L1_Pos_Controller::update_roll_setpoint()
 {
-	float ret = atanf(_lateral_accel * 1.0f / CONSTANTS_ONE_G);
-	ret = math::constrain(ret, -_roll_lim_rad, _roll_lim_rad);
-	return ret;
+	float roll_new = atanf(_lateral_accel * 1.0f / CONSTANTS_ONE_G);
+	roll_new = math::constrain(roll_new, -_roll_lim_rad, _roll_lim_rad);
+
+	// no slew rate limiting active
+	if (_dt <= 0.0f || _roll_slew_rate <= 0.0f) {
+		_roll_setpoint = roll_new;
+		return;
+	}
+
+	// slew rate limiting active
```

Comment:
Constrain to _roll_setpoint +- _roll_slew_rate * _dt?

Reply:
@dagar Done!